src/internal/pachsql/url.go

```diff
@@ -26,7 +26,16 @@ func ParseURL(x string) (*URL, error) {
 	}
 	port, err := strconv.Atoi(u.Port())
 	if err != nil {
-		return nil, errors.EnsureStack(errors.Wrapf(err, "parsing url port: %w"))
```

Comment:
If we do this, then we can't tell whether the port was not set.  I think it makes more sense to do this in the `*DSN` methods and handle the case where port is 0 separately.  That way we move some of the protocol specific logic out of here.  What do you think?

Reply:
As discussed, the desired logic is to default to `port = 0` if the user doesn't specify a port. We then check for this and default to specific port numbers in the DB specific DSN functions.

Reply:
I don't think this will give the behavior we want. I would expect port to be 0 whenever there is an error here.   We need to check `if u.Port() != ""`.  If no port was parsed from the url then don't call `strconv.Atoi` to try to parse it.  If there was a port parsed then try to parse it and always error if it fails.